### PR TITLE
Add dev-requirements.txt

### PR DESCRIPTION
**Pull request by goldsborough**
Creates a `dev-requirements.txt` file for developers to quickly run `pip install -r dev-requirements.txt`

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-r requirements.txt
+ninja
+flake8
+numpy
+scipy
+mkl
```
Comment (soumith) [inline dev-requirements.txt]: if you're in a conda env, ideally these come in from anaconda.

Comment (goldsborough) [reply dev-requirements.txt]: How? When I setup a new conda env (using `https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh`) and clone a new pytorch I always have to manually pip install them, and then I always forgot some so it all takes time

**Comment by SsnL** [comment]
README.md should be updated if we want to do this.

**Comment by zou3519** [comment]
I think you're missing
```
pyyaml
typing
```
for the code gen

**Comment by goldsborough** [comment]
@zou3519 it's `-r requirements.txt`

**Comment by goldsborough** [comment]
@soumith @apaszke any more thoughts?

**Comment by soumith** [comment]
no more thoughts. I feel a bit uneasy knowing that we might break the dev's conda setup, if they have a mishmash of either combo of mkl, how it interacts with other packages they have already installed....

**Comment by goldsborough** [comment]
We can leave out `mkl` and `mkl-devel` maybe?

**Comment by goldsborough** [comment]
@soumith sounds like a good reason not to have this file, might just end up confusing people

**Pull request closed without merging**